src/main/java/dao/FruitDaoCsvImpl.java

```diff
@@ -0,0 +1,41 @@
+package dao;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class FruitDaoCsvImpl implements FruitDao {
```

Comment:
Please let's follow SOLID and common mistakes. So, I recommend creating 5 separate services for this task: ReaderService - read data from file, WriterService - write data to file, ReportMakerService - prepare string report, TransactionParserService - parse every single row from file to create list of FruitTransaction. FruitShopService - process all list of FruitTransaction using strategy

Reply:
Also consider the following:
![image](https://user-images.githubusercontent.com/53354844/225003091-193515bd-4021-4838-a5b3-16e36ff4be4b.png)